WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -1130,6 +1144,30 @@ def add_result(pos, name)
     end
   end
 
+  describe 'POST #bookmark' do
+    let!(:user) { FactoryBot.create(:user) }
+    let!(:competition) { FactoryBot.create(:competition, :visible) }
+
+    context 'when signed in' do
+      before do
+        sign_in user
+      end
+
+      it 'bookmarks a competition' do
+        expect(user.is_bookmarked?(competition)).to eq false
+        get :bookmark, params: { id: competition.id }
```

Comment:
nit: Can we make this endpoint (and our unbookmark endpoint) only accept `POST`s? That feels more semantically correct to me than having a `GET` cause a side effect.

Reply:
Hmm, I didn't mean to make this allow `GET`s...

Ah, I see.  `config/routes.rb` only allows `POST`, but since the controller is behind that file, nothing in this test is checking what the allowed methods are.  I manually verified that the site doesn't accept `GET`s on this route.

I switched these to posts for good style.

Reply:
Oh that's weird... Thanks for the info!